crates/interfaces/src/p2p/full_block.rs

```diff
@@ -461,6 +473,46 @@ where
         Some(response)
     }
 
+    fn on_headers_response(&mut self, headers: WithPeerId<Vec<Header>>) {
+        let (peer, mut headers_falling) =
+            headers.map(|h| h.into_iter().map(|h| h.seal_slow()).collect::<Vec<_>>()).split();
+
+        // fill in the response if it's the correct length
+        if headers_falling.len() == self.count as usize {
```

Comment:
what happens if this condition is not met? does the client re-request the headers?

Reply:
yes,

this is a simplification, because we don't expect fragmented header responses, or rather don't want to deal with them